src/pages/Generate.tsx

```diff
@@ -73,56 +226,151 @@ export const Generate = () => {
+                  mb={5}
+                  w="50%"
+                >
+                  <option value="2022">2022</option>
+                  <option value="2023">2023</option>
+                  <option value="2024">2024</option>
+                </Select>
+                <FormLabel htmlFor="term">Term</FormLabel>
+                <Select
+                  placeholder="Select Term"
+                  value={term}
+                  onChange={(e) => setTerm(e.target.value)}
+                  mb={5}
+                  w="50%"
+                >
+                  <option value="SPRING">Spring</option>
+                  <option value="SUMMER">Summer</option>
+                  <option value="FALL">Fall</option>
+                </Select>
+                <FormLabel htmlFor="courses">Courses</FormLabel>
+                <SelectPlus<CourseOption, true, GroupBase<CourseOption>>
+                  isMulti
+                  name="courses"
+                  options={courseOptions}
+                  placeholder="Select courses"
+                  onChange={handleCourseChange}
+                />
+                <FormLabel mt={5} htmlFor="algorithm1">
+                  Algorithm 1
+                </FormLabel>
+                <RadioGroup
+                  id="algorithm1"
+                  name="algorithm1"
+                  onChange={setAlgorithm1}
+                  value={algorithm1}
+                >
+                  <Stack direction="row">
+                    <Radio value="COMPANY3">Company 3</Radio>
+                    <Radio value="COMPANY4">Company 4</Radio>
+                  </Stack>
+                </RadioGroup>
+                <FormLabel mt={5} htmlFor="algorithm2">
+                  Algorithm 2
+                </FormLabel>
+                <RadioGroup
+                  id="algorithm2"
+                  name="algorithm2"
+                  onChange={setAlgorithm2}
+                  value={algorithm2}
+                >
+                  <Stack direction="row">
+                    <Radio value="COMPANY3">Company 3</Radio>
+                    <Radio value="COMPANY4">Company 4</Radio>
+                  </Stack>
+                </RadioGroup>
+              </GridItem>
+              <GridItem
+                bg="gray.800"
+                p={10}
+                borderRadius={10}
+                flexDir="column"
+                style={{ boxShadow: "0px 0px 10px rgba(0, 0, 0, 0.40)" }}
+              >
+                <Heading mb={4} size="md">
+                  Course Selections
+                </Heading>
+                <Table variant="striped" size="sm">
+                  <Thead>
+                    <Tr>
+                      <Th>Course</Th>
+                      <Th>Sections</Th>
+                    </Tr>
+                  </Thead>
+                  <Tbody>
+                    {selectedCourses
+                      ? selectedCourses?.map((course) => (
```

Comment:
I think typescript should be smart enough for you to omit the `?` here and just put `selectedCourses`. If not you should change it to `selectedCourses!` since you previously asserted that it's not undefined